Framework/DataHandling/src/LoadNXcanSAS.cpp

```diff
@@ -246,238 +281,171 @@ void loadInstrument(H5::Group &entry, const Mantid::API::MatrixWorkspace_sptr &w
   }
 }
 
-// ----- DATA
-WorkspaceDimensionality getWorkspaceDimensionality(H5::Group &dataGroup) {
-  WorkspaceDimensionality dimensionality(WorkspaceDimensionality::other);
-  auto intensity = dataGroup.openDataSet(sasDataI);
-  auto dataSpaceInfo = getDataSpaceInfo(intensity);
-
-  if (dataSpaceInfo.dimSpectrumAxis == 1 && dataSpaceInfo.dimBin > 0) {
-    dimensionality = WorkspaceDimensionality::oneD;
-  } else if (dataSpaceInfo.dimSpectrumAxis > 1 && dataSpaceInfo.dimBin > 0) {
-    dimensionality = WorkspaceDimensionality::twoD;
-  } else {
-    dimensionality = WorkspaceDimensionality::other;
+//----- Data ----- //
+
+/**
+ * Prepares the slabShape and offset vectors to extract data from a given slice of the signal or Q datasets in the H5
+ * file. For non polarized data, slabShape and position indexes have same dimensions, but for polarized data these
+ * vectors have to be trimmed for extracting Q datasets.
+ * @param axesIndex: Index to know which kind of axes the data is going to be stored in the workspace (X, Y, XErr,
+ * YErr). If data is X or XErr, the dimensions of the slab and offset vectors are different.
```

Comment:
Could you add a short explanation to why this is the case to help future developers?
- Why vectors have to be trimmed for polarization
- Why X or XErr have different vector dims